src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/constraint/MonthYearConstraintValidator.kt

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.constraint
+
+import org.ocpsoft.prettytime.nlp.PrettyTimeParser
+import javax.validation.ConstraintValidator
+import javax.validation.ConstraintValidatorContext
+
+
+class MonthYearValidator : ConstraintValidator<ValidMonthYear?, String?> {
```

Comment:
I'm not sure what happens if the parse method bombs out. Just to be sure I've wrapped it in a runCatching:  
```
 override fun isValid(datesString: String?, context: ConstraintValidatorContext) =
            Result.runCatching { PrettyTimeParser().parse(datesString).size == 1}.getOrElse { false }
```

}